### DISCUSSION TITLE
Is that a way to start a single Application when client connected to xrdp server

### DISCUSSION THREAD

**Discussion by fuqifacai** [neutrinolabs/xrdp#3321] [General]
Is that a way to start a single Application instead of default Desktop or screen when client connected to xrdp server ?
And can send parameters from client to server side as RDP did.
Thanks for any advice.

**Comment by matt335672** [comment]
What you want is RAIL support. That's not fully implemented yet.

**Reply by fuqifacai** [reply]
In Microsoft 's RDP , we can run a single application remotely instead of remote windows desktop.
I just want the same feature in XRDP protocol.
Thanks.

**Comment by fuqifacai** [comment]
Or do we have workaround ? 
I can run a application which set in the startwm.sh , is there we can send or set some parameters to  it ?
![image](https://github.com/user-attachments/assets/eb7b145c-b14e-4241-b8a7-c57835dfa894)

**Reply by matt335672** [reply]
We can't pass parameters from xrdp.

If you know what the application is, you can possibly read something from a local file on the disk.

**Reply by fuqifacai** [reply]
Thanks reply. 
Read a local file sounds possible , but how the client to write it ?
I want to support several applications in the remote server , therefore I want the client could determine which remote application it will load via send this info by startup parameters , like Microsoft's RDP did.
Microsoft's RDP use a config file to tell server which application and parameters will be called.

**Reply by matt335672** [reply]
I can't see a way to do that at the moment I'm afraid.

**Reply by fuqifacai** [reply]
helpful , Thanks for your confirmation.
Really appreciated.
